algorithms/kernel/linear_regression/linear_regression_training_input.cpp

```diff
@@ -88,7 +88,14 @@ services::Status Input::check(const daal::algorithms::Parameter * par, int metho
     NumericTablePtr dataTable = get(data);
     size_t nRowsInData        = dataTable->getNumberOfRows();
     size_t nColumnsInData     = dataTable->getNumberOfColumns();
-    DAAL_CHECK(nRowsInData >= nColumnsInData + (int)(method == qrDense && parameter->interceptFlag == true), ErrorIncorrectNumberOfRows);
+    if (method == normEqDense)
+    {
+        DAAL_CHECK(nRowsInData >= nColumnsInData + (int)(parameter->interceptFlag == true), ErrorIncorrectNumberOfRows);
```

Comment:
nRowsInData == 0 and nColumnsInData == 0 is valid case for this check, if intercept was disabled

Reply:
The case of nRowsInData == 0 or nColumnsInData == 0 is checked in regression::training::Input::check, which is called on line #84 in this check.